internal/commands/app/create_inputs.go

```diff
@@ -268,6 +272,58 @@ func (i *createInputs) resolveClusters(ui terminal.UI, client atlas.Client, grou
 	return dsClusters, nonExistingClusters, nil
 }
 
+func (i *createInputs) resolveServerlessInstances(ui terminal.UI, client atlas.Client, groupID string) ([]dataSourceCluster, []string, error) {
+	if i.Template != "" && len(i.ServerlessInstances) > 0 {
+		return nil, nil, errors.New("cannot create a template app with Serverless instances")
+	}
+
+	serverlessInstances, err := client.ServerlessInstances(groupID)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	existingServerlessInstances := map[string]struct{}{}
+	for _, d := range serverlessInstances {
+		existingServerlessInstances[d.Name] = struct{}{}
+	}
+	nonExistingServerlessInstances := make([]string, 0, len(i.ServerlessInstances))
+
+	dsServerlessInstances := make([]dataSourceCluster, 0, len(i.ServerlessInstances))
+	for idx, serverlessInstanceName := range i.ServerlessInstances {
+		if _, ok := existingServerlessInstances[serverlessInstanceName]; !ok {
+			nonExistingServerlessInstances = append(nonExistingServerlessInstances, serverlessInstanceName)
+			continue
+		}
+
+		serviceName := serverlessInstanceName
+		if len(i.ServerlessInstanceServiceNames) > idx {
+			serviceName = i.ServerlessInstanceServiceNames[idx]
+		} else {
+			if !ui.AutoConfirm() {
+				if err := ui.AskOne(&serviceName, &survey.Input{
+					Message: fmt.Sprintf("Enter a Service Name for Serverless instance '%s'", serverlessInstanceName),
+					Default: serviceName,
+				}); err != nil {
+					return nil, nil, err
+				}
+			}
+		}
+		dsServerlessInstances = append(dsServerlessInstances,
+			dataSourceCluster{
```

Comment:
Since `dataSourceCluster` seems to be a representation of the mongodb-atlas service type, I think it makes sense to use it for serverless instances as well. I'm down to combine `dsClusters` and `dsServerlessInstances` and just have one for loop though!

Reply:
meh i think it's fine to leave as-is... was mostly curious to hear it from someone who's been more in the weeds with all of this stuff